File: SWIFT/Cap03/c03ex06.swift
// c03ex06.swift

import Foundation

func trocar(_ palavra: inout [Character], 
            _ posicao: Int, _ novoCaractere: Character) {
    palavra[posicao] = novoCaractere
}

func centro(_ palavra: [Character]) -> String {
    let tamanho = palavra.count
    let inicio = tamanho / 2 - 2
    let fim = tamanho / 2 + 1
    let segmento = palavra[inicio...fim]
    return String(segmento)
}

var palavra: [Character] = Array("COMPUTADOR")
let tamanho = palavra.count
var copias = ""

var novaPalavra = palavra
trocar(&novaPalavra, 6, "O")

print("Palavra ........: \(String(palavra))")
print("")
print("Lado esquerdo ..: \(String(palavra.prefix(3)))")
print("Centro .........: \(centro(palavra))")
print("Lado direito ...: \(String(palavra.suffix(3)))")
print("Reverso ........: \(String(palavra.reversed()))")
copias = String(palavra) + String(palavra) + String(palavra)
print("Copias .........: \(copias)")
print("Nova palavra ...: \(String(novaPalavra))")

print("")
print("Tecle <Enter> para encerrar... ", terminator: "")
_ = readLine()

File: VB_NET/Cap03/c03ex06.vb
' c03ex06.vb

Imports System
Imports System.Linq
Imports System.Console

Module Program
    Sub Main(args As String())
        Dim palavra As String, tamanho As Integer, copias As String
        Dim reverso As String, nvPalavra As String

        palavra = "COMPUTADOR"
        tamanho = palavra.Length
        copias = palavra & palavra & palavra
        reverso = New _
            String(palavra.ToCharArray().Reverse().ToArray())
        nvPalavra = _
            palavra.Substring(0, 6) & "O" & palavra.Substring(7)
            
        WriteLine($"Palavra ........: {palavra}")
        WriteLine()
        WriteLine($"Lado esquerdo ..: {palavra.Substring(0, 3)}")
        WriteLine($"Centro .........: {palavra.Substring(3, 4)}")
        WriteLine($"Lado direito ...: {palavra.Substring(tamanho _
            - 3)}")
        WriteLine($"Reverso ........: {reverso}")
        WriteLine($"Copias .........: {copias}")
        WriteLine($"Tira pedaco ....: {palavra.Substring(0, 7)}")
        WriteLine($"Nova palavra ...: {nvPalavra}")

        Write(vbCrLf + "Aperte <Enter> para encerrar... ")
        ReadLine()
    End Sub
End Module
   
